src/hooks/usePrefersColorScheme.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react'
+
+type ColorSchemeType = 'dark' | 'light' | 'no-preference'
+
+const usePrefersColorScheme = (): ColorSchemeType => {
+  const [preferredColorScheme, setPreferredColorScheme] = useState<ColorSchemeType>('no-preference')
+
+  useEffect(() => {
+    if (typeof window.matchMedia !== 'function') return
+
+    const isDark = window.matchMedia('(prefers-color-scheme: dark)')
+    const isLight = window.matchMedia('(prefers-color-scheme: light)')
+
+    const darkListener = ({ matches }: MediaQueryListEvent) => {
+      matches && setPreferredColorScheme('dark')
+    }
+    const lightListener = ({ matches }: MediaQueryListEvent) => {
+      matches && setPreferredColorScheme('light')
+    }
```

Comment:
You could curry these to reduce repetition, e.g.

```suggestion
    const listener = (type: ColorSchemeType) = ({ matches }: MediaQueryListEvent) => {
      matches && setPreferredColorScheme(type)
    }
    
    const darkListener = listener('dark')
    const lightListener = listener('light')
```

Reply:
Good suggestion! Done